packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -21,7 +21,9 @@ GenSnapshot get genSnapshot => context.putIfAbsent(GenSnapshot, () => const GenS
 
 /// A snapshot build configuration.
 class SnapshotType {
-  const SnapshotType(this.platform, this.mode);
+  SnapshotType(this.platform, this.mode) {
```

Comment:
Any reason to not keep the const? (and if not, `s/new/const/` in `buildScriptSnapshot`)

Reply:
Ah yep - not sure how many times that's bit me now! The engine is compiled with that flag set, but we don't run the tool that way. Thanks.

Reply:
Not sure about `const`, but simple fix for asserts support in `pub run test` is to set `DART_VM_OPTIONS=--assert-initializer` for `pub run test` invocation.